### PR TITLE
Feature/header tabs

### PR DESCRIPTION
**Pull request by radireddy**
Added header tabs component

Staging and other details in below jira task number
UU-838

### CODE REVIEW

File: packages/oui-header-tabs/src/header-tab.directive.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export default () => ({
+    template,
+    controller,
+    replace: true,
+    restrict: "E",
+    require: ["^^ouiHeaderTabs", "ouiHeaderTab"],
+    controllerAs: "$ctrl",
+    bindToController: true,
+    scope: {
+        state: "@",
```
Comment (AxelPeter) [inline packages/oui-header-tabs/src/header-tab.directive.js]: No `href` attribute ? Not all managers use states. 
The tabs components must be thought for a general use.

File: packages/oui-header-tabs/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    </oui-header-tab>
+    <oui-header-tab
+        data-text="Tab13"
+        data-state="showcase.oui-angular.header-tabs.tab13">
+    </oui-header-tab>
+</oui-header-tabs>
+<div class="oui-doc-preview-only-keep-children" style="margin-bottom: 15px;">
+```
+
+
```
Comment (AxelPeter) [inline packages/oui-header-tabs/README.md]: Useless extra spaces

File: packages/oui-header-tabs/src/header-tab.directive.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import _ from "lodash";
+import controller from "./header-tab.controller";
+import template from "./header-tab.html";
+
+export default () => ({
```
Comment (AxelPeter) [inline packages/oui-header-tabs/src/header-tab.directive.js]: Must be a component, not a directive

File: packages/oui-header-tabs/src/header-tab.directive.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    controllerAs: "$ctrl",
+    bindToController: true,
+    scope: {
+        state: "@",
+        stateParams: "<?",
+        text: "@",
+        disabled: "<?"
+    },
+    transclude: true,
+    link: ($scope, $element, $attrs, $ctrls) => {
```
Comment (AxelPeter) [inline packages/oui-header-tabs/src/header-tab.directive.js]: Must be in a class

File: packages/oui-header-tabs/src/header-tab.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<li class="oui-tabs__tab"
+    data-ng-class="{ 'oui-tabs__tab_active': $ctrl.active, 'oui-tabs__tab_animate': $ctrl.isActivating }">
+    <a data-ng-if="$ctrl.state && !$ctrl.disabled"
+        aria-expanded="{{$ctrl.active ? 'true' : 'false' }}"
+        ui-sref="{{ $ctrl.state + '(' + String($ctrl.stateParams) + ')' }}">
+        {{$ctrl.text}}
+    </a>
+    <a data-ng-if="$ctrl.state && $ctrl.disabled" class="oui-tabs__tab_disabled">
```
Comment (AxelPeter) [inline packages/oui-header-tabs/src/header-tab.html]: Should be `span` if it's a link without `href`.
The modifier class should be with `oui-tabs__tab` on the `li`. We don't use modifier class alone

File: packages/oui-header-tabs/src/header-tabs.component.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import controller from "./header-tabs.controller";
+import template from "./header-tabs.html";
+
+export default {
+    template,
+    controller,
+    transclude: true,
+    restrict: "E",
```
Comment (AxelPeter) [inline packages/oui-header-tabs/src/header-tabs.component.js]: No need for component

File: packages/oui-header-tabs/src/header-tabs.component.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+    controllerAs: "$ctrl"
```
Comment (AxelPeter) [inline packages/oui-header-tabs/src/header-tabs.component.js]: No need too, it's the default value :)

File: packages/oui-header-tabs/src/header-tabs.html [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<nav role="navigation">
+    <ul class="oui-tabs"
+        role="tablist"
+        data-ng-transclude>
```
Comment (AxelPeter) [inline packages/oui-header-tabs/src/header-tabs.html]: No `data-*` prefixing in our templates ;)

File: packages/oui-header-tabs/src/header-tabs.service.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        this.registeredTabs = [];
+        this.activeTab = undefined;
+
+        this.onTransitionHandler = $transitions.onSuccess({}, () => {
+            this.refreshActiveTab();
+        });
+    }
+
+    $onInit () {
+        this.init();
```
Comment (AxelPeter) [inline packages/oui-header-tabs/src/header-tabs.service.js]: Why not use directly `$onInit` ?

File: packages/oui-header-tabs/src/header-tabs.service.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
+import _ from "lodash";
+
+export default class HeaderTabsService {
+    constructor ($state, $transitions) {
+        "ngInject";
+        this.$state = $state;
+        this.registeredTabs = [];
```
Comment (AxelPeter) [inline packages/oui-header-tabs/src/header-tabs.service.js]: Is already on init()

File: packages/oui-header-tabs/src/header-tabs.controller.js [hunk 11]
```diff
@@ -1,20 +1,59 @@
+import _ from "lodash";
```
Comment (FredericEspiau) [inline packages/oui-header-tabs/src/header-tabs.controller.js]: You could only import the `_.prototype.find` function

File: packages/oui-header-tabs/src/header-tabs.controller.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+            this.onActivate({ tab: tab.text });
+        }
+    }
+
+    onInactivateHandler (tab) {
+        if (this.onInactivate) {
+            this.onInactivate({ tab: tab.text });
+        }
+    }
+
```
Comment (FredericEspiau) [inline packages/oui-header-tabs/src/header-tabs.controller.js]: Extra whitespace

File: packages/oui-header-tabs/src/header-tabs.controller.js [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    onActivateHandler (tab) {
+        if (this.onActivate) {
+            this.onActivate({ tab: tab.text });
+        }
+    }
+
+    onInactivateHandler (tab) {
+        if (this.onInactivate) {
+            this.onInactivate({ tab: tab.text });
```
Comment (FredericEspiau) [inline packages/oui-header-tabs/src/header-tabs.controller.js]: The name of the property is `tab` but its content is the text (name I guess) of the tab ? Then the property should be named `tabName` for example

File: packages/oui-header-tabs/src/header-tabs.controller.js [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+    addTab (tab) {
+        this.tabs.push(tab);
+        if (!this.activeTab) {
+            this.updateActiveTab(tab);
+        }
+    }
+
+    unselectTab (tab) {
+        this.activeTab = null;
```
Comment (FredericEspiau) [inline packages/oui-header-tabs/src/header-tabs.controller.js]: Only one tab is active at the same time ? Then you don't need an argument

File: packages/oui-header-tabs/src/header-tabs.controller.js [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    $onInit () {
+        this.tabs = [];
+        this.activeTab = null;
+    }
+
+    updateActiveTab () {
+        const currentTab = _.find(this.tabs, tab => tab.state && this.$state.includes(tab.state));
+        if (currentTab && currentTab !== this.activeTab) {
```
Comment (FredericEspiau) [inline packages/oui-header-tabs/src/header-tabs.controller.js]: Isn't it the same as `currentTab !== this.activeTab` ?

If `currentTab === undefined` then we go to the `else if` either way (`this.activeTab` can't be `undefined`)

Also, you are comparing objects, it feels dangerous to me but I don't know, I might be paranoid but I would have methods to:
* determine if a tab is currently active
* determine if two tabs are equal

File: packages/oui-header-tabs/src/header-tab.controller.js [hunk 16]
```diff
@@ -0,0 +1,29 @@
+import { addBooleanParameter } from "@oui-angular/common/component-utils";
+
+export default class {
+    constructor ($timeout, $attrs) {
+        "ngInject";
+        this.$timeout = $timeout;
+        this.$attrs = $attrs;
+        this.ANIMATION_TIME = 200;
```
Comment (FredericEspiau) [inline packages/oui-header-tabs/src/header-tab.controller.js]: :+1:

File: packages/oui-header-tabs/src/header-tabs.controller.js (hunk 11 above)
Comment (radireddy) [reply packages/oui-header-tabs/src/header-tabs.controller.js]: 👍

File: packages/oui-header-tabs/src/header-tabs.controller.js (hunk 15 above)
Comment (radireddy) [reply packages/oui-header-tabs/src/header-tabs.controller.js]: both of these can be undefined, in that case selectTab must not be called

File: packages/oui-header-tabs/README.md [hunk 17]
```diff
@@ -0,0 +1,199 @@
...
+# Header Tabs
+<component-status cx-design="complete" ux="complete"></component-status>
+## Usage
```
Comment (neolitec) [inline packages/oui-header-tabs/README.md]: Blank line between titles ;)

File: packages/oui-header-tabs/src/header-tab-content-transclude.directive.js [hunk 18]
```diff
@@ -0,0 +1,27 @@
+export default () => {
```
Comment (neolitec) [inline packages/oui-header-tabs/src/header-tab-content-transclude.directive.js]: Is there a reason this is not a class here, like everywhere in this project?

File: packages/oui-header-tabs/src/header-tab-content-transclude.directive.js [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+    "ngInject";
+    return {
+        restrict: "A",
+        require: "^ouiHeaderTabs",
+        link: (scope, elm, attrs) => {
+            const tab = scope.$eval(attrs.ouiTabContentTransclude);
+            tab.$transcludeFn((contents) => {
```
Comment (neolitec) [inline packages/oui-header-tabs/src/header-tab-content-transclude.directive.js]: Oww. I think we have to sit down all together and discuss about the tabs before simply copy paste... the past and doing weird things.

**Comment by AxelPeter** [comment]
The component is now prioritized by CX and will be taken into account by the UI Kit team
Thanks for your contribution !

**Pull request closed without merging**